scratchapi2/api.py

```diff
@@ -0,0 +1,33 @@
+"""Contains the APIClass and APISingleton base classes."""
+import requests
+
+class APIClass(object):
+    """Base class for classes that access the API."""
+
+    def __init__(self, api_url="https://api.scratch.mit.edu/"):
+        """Initialize the object with the API URL."""
+        self.api_url = api_url
+
+    def __repr__(self):
+        """Represent the object."""
+        return "<{}>".format(type(self).__name__)
+
+    __str__ = __repr__
+
+    def _request(self, path, *opts, api_url=None):
+        """Internal method to request data from the API."""
+        return requests.get(
+            (api_url or self.api_url)
+            + path.format(*opts)
+        ).json()
+
+class APISingleton(APIClass):
```

Comment:
The singleton logic is split off in case you decide to add a non-singleton `APIClass`.